sound/soc/sof/intel/hda.c

```diff
@@ -1478,18 +1499,22 @@ struct snd_soc_acpi_mach *hda_machine_select(struct snd_sof_dev *sdev)
 	mach = snd_soc_acpi_find_machine(desc->machines);
 	if (mach) {
 		bool add_extension = false;
+		bool tplg_fixup = false;
 
 		/*
 		 * If tplg file name is overridden, use it instead of
 		 * the one set in mach table
 		 */
-		if (!sof_pdata->tplg_filename)
+		if (!sof_pdata->tplg_filename) {
 			sof_pdata->tplg_filename = mach->sof_tplg_filename;
+			tplg_fixup = true;
+		}
```

Comment:
Do we need similar change in `atom.c`?

Reply:
no, we don't have the ability to override the topology filename in sof-acpi-dev.c. We only added this topology override support for PCI.

Reply:
only the fw_path and tplg_path parameters are exposed. We haven't seen the need to go further, and with SOF/ACPI support in "won't fix" mode there's no real incentive to do better.